File: Quick-runtime/src/main/java/com/wpf/app/quick/runtime/Databinder.java
package com.wpf.app.quick.runtime;

import androidx.annotation.Nullable;

import java.lang.reflect.Field;

/**
 * 可以获取参数值
 */
public interface Databinder extends Unbinder {

  default @Nullable
  Object getFieldValue(String fieldName) {
    try {
      Field findF = getClass().getDeclaredField(fieldName);
      findF.setAccessible(true);
      return findF.get(this);
    } catch (IllegalAccessException | NoSuchFieldException e) {
      e.printStackTrace();
    }
    return null;
  }
}

File: Quick-annotations/src/main/java/com/wpf/app/quick/annotations/bind/Databinder.kt
package com.wpf.app.quick.annotations.bind

interface Databinder {

    fun getFieldValue(fieldName: String): Any? {
        try {
            val findF = javaClass.getDeclaredField(fieldName)
            findF.isAccessible = true
            return findF[this]
        } catch (e: IllegalAccessException) {
            e.printStackTrace()
        } catch (e: NoSuchFieldException) {
            e.printStackTrace()
        }
        return null
    }
}